Fix multiple clicks on add to cart freezes page

rdy4ever opened the pull request:
<!--
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
 -->

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | As stated in #9634 and  #15483, if a user clicks on "add to cart" multiple times (and fast enough), the modal opens and closes multiple times, and then it "freezes", meaning you can no longer close it by clicking on "Continue shopping" or the close button. This also creates the undesired effect of adding to cart more products then intended.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature?no
| Fixed ticket? | Fixes #9634 Fixes #15483
| How to test?  | See before/after screen rec: https://drive.google.com/open?id=1PMgk5geCcQj0LGaTB-VYCFbQAkos4MAH
<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/15760)
<!-- Reviewable:end -->

rdy4ever committed:
Trim unnecesary white spaces from customer name

Remove whitespace from both sides of the customer name when input loses focus.

rdy4ever committed:
Revert "Trim unnecesary white spaces from customer name"

This reverts commit 1bfdc4b0f3a2406d4fdc9ca3e0a5026580882ad9.

rdy4ever committed:
Merge pull request #1 from PrestaShop/develop

Getting up to date

rdy4ever committed:
Merge branch 'develop' of https://github.com/rdy4ever/PrestaShop into develop

rdy4ever committed:
Merge pull request #2 from PrestaShop/develop

Getting up to date

rdy4ever committed:
Merge branch 'develop' of https://github.com/rdy4ever/PrestaShop into develop

rdy4ever committed:
FIx multiple add to cart freezes modal

rdy4ever committed:
Rebuild assets

PierreRambaud commented on themes/_core/js/cart.js:
```diff
@@ -79,9 +79,9 @@ $(document).ready(() => {
-          $('[data-button-action="add-to-cart"]').attr('disabled', 'disabled');
-      } else {
+      if ($('#quantity_wanted').val() <= $('[data-stock]').data('stock') && $('[data-allow-oosp]').data('allow-oosp').length !== 0) {
+        $('[data-button-action="add-to-cart"]').attr('disabled', 'disabled');
```
```suggestion
        $('[data-button-action="add-to-cart"]').prop('disabled', true);
```

rdy4ever replied:
Sure, done

PierreRambaud commented on themes/_core/js/cart.js:
```diff
@@ -126,6 +126,7 @@ $(document).ready(() => {
           });
         }).fail((resp) => {
           prestashop.emit('handleError', {eventType: 'addProductToCart', resp: resp});
+          $('[data-button-action="add-to-cart"]').removeAttr('disabled');
```
```suggestion
          $('[data-button-action="add-to-cart"]').removeProp('disabled');
```

PierreRambaud commented:
Could you please squash your commit and update your develop branch when you create a branch, it generates unnecessary commits history :sweat_smile:

rdy4ever committed:
prop

rdy4ever committed:
rebuild assets

rdy4ever commented:
> Could you please squash your commit and update your develop branch when you create a branch, it generates unnecessary commits history 😅

I'm a bit shamed by this... I'm on gitHub Desktop on Windows and have no ideea how to squash :))

PierreRambaud commented:
@rdy4ever I only know how to do it in CLI, I'm not a UI user :sweat_smile: 
I can do it for you if you want.

rdy4ever commented:
> @rdy4ever I only know how to do it in CLI, I'm not a UI user 😅
> I can do it for you if you want.

Sure, go ahead

rdy4ever commented:
Seems `data-stock` and `data-allow-oos` are not available on quick view and products without stock. I need to check things out, for now this is not reliable.

rdy4ever committed:
Remove obsolete condition

We don't need this condition as we disable the button in all cases... duh :))

rdy4ever committed:
Reformat this function

Had to reformat the code this whole function due to eliminating the condition... please see previous commit for clearer diff

rdy4ever committed:
Regenerate assets

rdy4ever commented:
@PierreRambaud 
Just realised we don't need the previous condition at all, because we disable the button anyway... duh :))
Terribly sorry for reformatting the whole function (I know this makes it harder to review the diff), but, with the condition out of the way, the indenting was plain wrong.
Could you please review and squash my many commits as possible? Thanks!

PierreRambaud commented:
Hi @rdy4ever 
Is this pull request still needed?

rdy4ever commented:
Well, it's at least outdated. I'll close it and maybe redoit sometime.
